app/services/clients/stripe/stripePerson.model.test.js

```diff
@@ -0,0 +1,222 @@
+    expect(() => new StripePerson({
+      first_name: null,
+      last_name: lastName,
+      address_line1: addressLine1,
+      address_city: addressCity,
+      address_postcode: addressPostcode,
+      dob_day: dobDay,
+      dob_month: dobMonth,
+      dob_year: dobYear
+    })).to.throw('StripePerson "first_name" must be a string')
+  })
+
+  it('should fail when property that should be integer is a string that coerces to an integer', () => {
+    expect(() => new StripePerson({
+      first_name: firstName,
+      last_name: lastName,
+      address_line1: addressLine1,
+      address_city: addressCity,
+      address_postcode: addressPostcode,
+      dob_day: '8',
+      dob_month: dobMonth,
+      dob_year: dobYear
+    })).to.throw('StripePerson "dob_day" must be a number')
+  })
+
+  it('should fail when property that should be integer is number but not integer', () => {
+    expect(() => new StripePerson({
+      first_name: firstName,
+      last_name: lastName,
+      address_line1: addressLine1,
+      address_city: addressCity,
+      address_postcode: addressPostcode,
+      dob_day: 8.1,
+      dob_month: dobMonth,
+      dob_year: dobYear
+    })).to.throw('StripePerson "dob_day" must be an integer')
+  })
+
+  it('should fail when property that should be integer is null', () => {
+    expect(() => new StripePerson({
+      first_name: firstName,
+      last_name: lastName,
+      address_line1: addressLine1,
+      address_city: addressCity,
+      address_postcode: addressPostcode,
+      dob_day: null,
+      dob_month: dobMonth,
+      dob_year: dobYear
+    })).to.throw('StripePerson "dob_day" must be a number')
+  })
+
+  it('should fail when day is less than 1', () => {
+    expect(() => new StripePerson({
+      first_name: firstName,
+      last_name: lastName,
+      address_line1: addressLine1,
+      address_city: addressCity,
+      address_postcode: addressPostcode,
+      dob_day: 0,
+      dob_month: dobMonth,
+      dob_year: dobYear
+    })).to.throw('StripePerson "dob_day" must be larger than or equal to 1')
+  })
+
+  it('should fail when day is more than 31', () => {
+    expect(() => new StripePerson({
+      first_name: firstName,
+      last_name: lastName,
+      address_line1: addressLine1,
+      address_city: addressCity,
+      address_postcode: addressPostcode,
+      dob_day: 32,
+      dob_month: dobMonth,
+      dob_year: dobYear
+    })).to.throw('StripePerson "dob_day" must be less than or equal to 31')
+  })
+
+  it('should fail when month is less than 1', () => {
+    expect(() => new StripePerson({
+      first_name: firstName,
+      last_name: lastName,
+      address_line1: addressLine1,
+      address_city: addressCity,
+      address_postcode: addressPostcode,
+      dob_day: dobDay,
+      dob_month: 0,
+      dob_year: dobYear
+    })).to.throw('StripePerson "dob_month" must be larger than or equal to 1')
+  })
+
+  it('should fail when month is less than 1000', () => {
```

Comment:
probably copy pasta `should fail when year is less than 1000` ?

Reply:
I never copy and paste, Dan. I type everything out by hand. Your eyes are deceiving you.

Reply:
No, really, should be fixed now and also added a test for the month being too large.

Reply:
:D